otj-kafka-spring/src/main/java/com/opentable/kafka/spring/builders/KafkaFactoryBuilderFactoryBean.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.opentable.kafka.spring.builders;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.codahale.metrics.MetricRegistry;
+
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import com.opentable.kafka.builders.EnvironmentProvider;
+import com.opentable.kafka.metrics.OtMetricsReporterConfig;
+import com.opentable.kafka.util.ClientIdGenerator;
+import com.opentable.service.ServiceInfo;
+import com.opentable.spring.PropertySourceUtil;
+
+/**
+ * Main spring entry point for building KafkaConsumer and KafkaProducer
+ */
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class KafkaFactoryBuilderFactoryBean {
+
+    protected static final String PREFIX = "ot.kafka.";
+    static final String DEFAULT = "default";
+
+    private final ConfigurableEnvironment env;
+    final Optional<MetricRegistry> metricRegistry;
+    final EnvironmentProvider environmentProvider;
+    final Optional<ServiceInfo> serviceInfo;
+    private final String consumerPrefix;
+    private final String producerPrefix;
+
+    KafkaFactoryBuilderFactoryBean(
```

Comment:
Why is this being copied to otj-kafka-spring?

Reply:
Hm.. This one provides SpringKafkaProducerFactoryBuilder and SpringKafkaConsumerFactoryBuilder but I agree it looks pretty similar. Any Ideas? (Inheritance with beans is a pain, in my opinion)

Reply:
Did you ever reexamine this? It still looks fairly ugly